src/modules/admin/sections/ProfileEditor/EditableProfile/EditableProfile.Avatar.tsx

```diff
@@ -1,19 +1,96 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
-
-import { Button } from '@src/common/components';
-import { Avatar, HStack } from '@chakra-ui/react';
+import { Avatar, Button } from '@src/common/components';
+import { useUser } from '@common/contexts';
+import { useUpdateAvatarApi } from '@src/services';
+import {
+  HStack,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+} from '@chakra-ui/react';
+import axios from 'axios';
+import { isNil } from 'lodash';
 
 export default function ProfileEditorAvatar() {
   const { t } = useTranslation('admin');
+  const { isOpen, onOpen, onClose } = useDisclosure();
```

Comment:
There's actually a custom hook `useToggle` that provides the same functionality. But this works too. 👍